src/utils/include/com/amazonaws/kinesis/video/utils/Include.h

```diff
@@ -1375,6 +1375,65 @@ PUBLIC_API SIZE_T getInstrumentedTotalAllocationSize();
 #define RESET_INSTRUMENTED_ALLOCATORS()             resetInstrumentedAllocatorsNoop()
 #endif
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+// File logging functionality
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * File logger error values starting from 0x41300000
+ */
+#define STATUS_FILE_LOGGER_BASE                                             STATUS_UTILS_BASE + 0x01200000
+#define STATUS_FILE_LOGGER_INDEX_FILE_INVALID_SIZE                          STATUS_SEMAPHORE_BASE + 0x00000001
+
+/**
+ * File based logger limit constants
+ */
+#define MAX_FILE_LOGGER_STRING_BUFFER_SIZE          (3 * 1024 * 1024)
+#define MIN_FILE_LOGGER_STRING_BUFFER_SIZE          (10 * 1024)
+#define MAX_FILE_LOGGER_LOG_FILE_COUNT              (10 * 1024)
+
+/**
+ * Default values used in the file logger
+ */
+#define FILE_LOGGER_LOG_FILE_NAME                   "kvsFileLog"
+#define FILE_LOGGER_LAST_INDEX_FILE_NAME            "kvsFileLogIndex"
+#define FILE_LOGGER_STRING_BUFFER_SIZE              (100 * 1024)
+#define FILE_LOGGER_LOG_FILE_COUNT                  3
+#define FILE_LOGGER_LOG_FILE_DIRECTORY_PATH         "./"
+
+/**
+ * Creates a file based logger object and installs the global logger callback function
+ *
+ * @param - UINT64 - IN - Size of string buffer in file logger. When the string buffer is full the logger will flush everything into a new file
+ * @param - UINT64 - IN - Max number of log file. When exceeded, the oldest file will be deleted when new one is generated
+ * @param - PCHAR - IN - Directory in which the log file will be generated
+ * @param - BOOL - IN - Whether to print log to std out too
+ * @param - BOOL - IN - Whether to set global logger function pointer
+ * @param - logPrintFunc* - OUT/OPT - Optional function pointer to be returned to the caller that contains the main function for actual output
+ *
+ * @return - STATUS code of the execution
+ */
+PUBLIC_API STATUS createFileLogger(UINT64, UINT64, PCHAR, BOOL, BOOL, logPrintFunc*);
+
+/**
+ * Frees the static file logger object and resets the global logging function if it was
+ * previously set by the create function.
+ *
+ * @return - STATUS code of the execution
+ */
+PUBLIC_API STATUS freeFileLogger();
+
+/**
+ * Helper macros to be used in pairs at the application start and end
+ */
+#define CREATE_DEFAULT_FILE_LOGGER()                                \
```

Comment:
I would prefer SET_XXX and RESET_XXX similar to other macros we use.

Reply:
I dont think SET_XXX is properly convey the purpose of this api. All this api does is create a file logger with default configs.

Reply:
Yeah, it creates and sets. I just wanted to point out the consistency.  Want to make sure that the setup and cleanup are in sync with each other and other declarations we had before. Your call.